utils/include/CRequestHandler.cpp

```diff
@@ -0,0 +1,154 @@
+#include "stdafx.h"
+
+#include "ERequestType.h"
+#include "CRequestExcAllData.h"
+#include "CRequestExcDiskData.h"
+#include "CRequestExcProcessData.h"
+#include "CDataProvider.h"
+#include "GlobalVariable.h"
+#include "CResponseFrame.h"
+#include "EFrameError.h"
+#include "Log.h"
+#include "CRequestHandler.h"
+
+CRequestHandler::CRequestHandler(
+    std::shared_ptr<IInfoDatabase> p_processes_data,
+    std::shared_ptr<IInfoDatabase> p_disks_data,
+    std::shared_ptr<IInfoDatabase> p_resources_data) : CInteractionHandler(),
+         m_data_base(p_processes_data, p_disks_data, p_resources_data)
+{ }
+
+CRequestHandler::CRequestHandler(const CRequestHandler& orig) : 
+    m_data_base(orig.m_data_base)
+{
+}
+
+bool CRequestHandler::HandleRequest(const std::string& request_str, 
+    std::string& answer)
+{
+    nlohmann::json request = nlohmann::json::parse(request_str);
+    CResponseFrame response(request[GlobalVariable::c_request_key_id]);
+    CLOG_DEBUG_START_FUNCTION();
+    try
+    {
+
+        if (!TryValidateRequest(request_str))
+        {
+            response.TryFormateResponse(answer, "",
+                EFrameError::INCORRECT_REQUEST); 
+            //log and return true
+	    	CLOG_ERROR("Invalid request from the client!!!");
+            return false;
+        }
+        EFrameError error_from_frame = GetErrorCodeFromFrame(request_str);
+        if (EFrameError::NONE != error_from_frame)
+        {
+            response.TryFormateResponse(answer, "",
+                error_from_frame);
+            //log and return true
+            CLOG_ERROR("Invalid request from the client!!!");
+            return false;
+        }
+
+        switch (AnalyzeRequestType(request))
+        {
+        case ERequestType::ALL_DATA:
+            if (!ExecuteRequest(answer, std::make_shared<CRequestExcAllData>(
```

Comment:
Create a temporary variable to create a request object in "switch case".
ExecuteRequest(answer, "temporary") after switch case

Reply:
Changed